src/wp-admin/includes/class-wp-list-table.php

```diff
@@ -602,18 +602,31 @@ protected function months_dropdown( $post_type ) {
 			$extra_checks = $wpdb->prepare( ' AND post_status = %s', $_GET['post_status'] );
 		}
 
-		$months = $wpdb->get_results(
-			$wpdb->prepare(
-				"
-			SELECT DISTINCT YEAR( post_date ) AS year, MONTH( post_date ) AS month
-			FROM $wpdb->posts
-			WHERE post_type = %s
-			$extra_checks
-			ORDER BY post_date DESC
-		",
-				$post_type
-			)
-		);
+		/**
+		 * Filters whether to short-circuit performing the months dropdown query.
+		 *
+		 * @since 5.7.0
+		 *
+		 * @param bool   $skip_query Whether to skip doing the query. Default false.
+		 * @param string $post_type  The post type.
+		 */
+		$skip_query = apply_filters( 'pre_months_dropdown_query', false, $post_type );
```

Comment:
Hm, so I was expecting this to return the list of months if they want to short-circuit the filter. Like we do in `get_option` or `get_blogs_of_user` for instance.

Reply:
There's an existing filter just below for filtering the results:

```
$months = apply_filters( 'months_dropdown_results', $months, $post_type );
```

The new filter short-circuits to skip the query and the existing filter allows you to filter the months.

What do you think @TimothyBJacobs ?

Reply:
A preemptive filter already exists to disable the dropdown completely. It's a guard clause at the opening of the method:

https://github.com/WordPress/wordpress-develop/blob/ea4841d30e40e3f049ca3403a480bb6e10420c89/src/wp-admin/includes/class-wp-list-table.php#L583-L596

Reply:
Re-reading the ticket's proposed enhancement, it seeks to give a way to skip the months query and allow a filtered months instead similar to https://core.trac.wordpress.org/browser/tags/5.5.1/src/wp-includes/media.php#L4106

Reply:
Agreed that is what they are looking for. If we are following that pattern, it'd use the results returned from the pre-filter.

Reply:
So you're proposing the new filter shifts the building of the dropdown's `label` and `select` to the 3rd party. Right?

Reply:
Sorry for the confusion @TimothyBJacobs. 

- Are you suggesting that it bails out and returns the dropdown HTML?
If yes, then we shifting building the dropdown's `label` and `select` to the 3d party callback. Is this what we want here?

- Or that it's used to filter the months and uses the returned months (if any is returned)? 
If yes, a filter already exists for filtering the months. 

https://github.com/WordPress/wordpress-develop/blob/b6a04b3b28307bdedd34395b56cb140536257775/src/wp-admin/includes/class-wp-list-table.php#L618-L626

Reply:
The second one, but the results filter shouldn't apply in that case. Having a double filter there is pretty typical.

Reply:
Makes sense. Captured in commit d500f96be781c33858dd7a35c216bef834875db0.

Reply:
LGTM!

Reply:
@TimothyBJacobs Thanks for your guidance (and patience)!